src/compiler/checker.ts

```diff
@@ -4583,7 +4583,22 @@ namespace ts {
                 let requireOptionalProperties = relation === subtypeRelation && !(source.flags & TypeFlags.ObjectLiteral);
                 for (let targetProp of properties) {
                     let sourceProp = getPropertyOfType(source, targetProp.name);
-                    if (sourceProp !== targetProp) {
+
+                    if (sourceProp === targetProp) { // source inherits targetProp and doesn't redeclare/override it.

```

Comment:
This test took place to make sure that derived classes of a class containing an abstract method "inherit" that method correctly -- either implementing the abstract method or else declaring the class abstract (and possibly redeclaring the method or adding overloads).

This was not the right place to perform this test since it's relevant to class-checking only.

Moved the test from `propertiesRelatedTo()` to `checkKindsOfPropertyMemberOverride()`.